bgpd/rfapi/bgp_rfapi_cfg.c

```diff
@@ -3864,489 +3864,469 @@ int bgp_rfapi_cfg_write(struct vty *vty, struct bgp *bgp)
-					vty_out(vty,
-						"  export %s%s prefix-list %s\n",
-						(rfg->type == RFAPI_GROUP_CFG_VRF ? "" : "zebra "),
-						afistr,
-						rfg->plist_export_zebra_name
-							[afi]);
-				}
-				/*
-				 * currently we only support redist plists for
-				 * bgp-direct.
-				 * If we later add plist support for
-				 * redistributing other
-				 * protocols, we'll need to loop over protocols
-				 * here
-				 */
-				if (rfg->plist_redist_name
-					    [ZEBRA_ROUTE_BGP_DIRECT][afi]) {
-					vty_out(vty,
-						"  redistribute bgp-direct %s prefix-list %s\n",
-						afistr,
-						rfg->plist_redist_name
-							[ZEBRA_ROUTE_BGP_DIRECT]
-							[afi]);
-				}
-				if (rfg->plist_redist_name
-					    [ZEBRA_ROUTE_BGP_DIRECT_EXT][afi]) {
-					vty_out(vty,
-						"  redistribute bgp-direct-to-nve-groups %s prefix-list %s\n",
-						afistr,
-						rfg->plist_redist_name
-							[ZEBRA_ROUTE_BGP_DIRECT_EXT]
-							[afi]);
-				}
+			if (rfg->rt_export_list) {
+				char *b = ecommunity_ecom2str(
+					rfg->rt_export_list,
+					ECOMMUNITY_FORMAT_ROUTE_MAP,
+					ECOMMUNITY_ROUTE_TARGET);
+				vty_out(vty, "  rt export %s\n", b);
+				XFREE(MTYPE_ECOMMUNITY_STR, b);
 			}
+		}
 
-			if (rfg->routemap_export_bgp_name) {
-				vty_out(vty, "  export %sroute-map %s\n",
-					(rfg->type == RFAPI_GROUP_CFG_VRF ? "" : "bgp "),
-					rfg->routemap_export_bgp_name);
+		/*
+		 * route filtering: prefix-lists and route-maps
+		 */
+		for (afi = AFI_IP; afi < AFI_MAX; ++afi) {
+			const char *afistr =
+				(afi == AFI_IP) ? "ipv4" : "ipv6";
+
+			if (rfg->plist_export_bgp_name[afi]) {
+				vty_out(vty,
+					"  export %s%s prefix-list %s\n",
+					(rfg->type == RFAPI_GROUP_CFG_VRF
+					 ? "" : "bgp "),
+					afistr,
+					rfg->plist_export_bgp_name[afi]);
 			}
-			if (rfg->routemap_export_zebra_name) {
-				vty_out(vty, "  export %sroute-map %s\n",
-					(rfg->type == RFAPI_GROUP_CFG_VRF ? "" : "zebra "),
-					rfg->routemap_export_zebra_name);
+			if (rfg->plist_export_zebra_name[afi]) {
+				vty_out(vty,
+					"  export %s%s prefix-list %s\n",
+					(rfg->type == RFAPI_GROUP_CFG_VRF
+					 ? "" : "zebra "),
+					afistr,
+					rfg->plist_export_zebra_name[afi]);
 			}
-			if (rfg->routemap_redist_name[ZEBRA_ROUTE_BGP_DIRECT]) {
+			/*
+			 * currently we only support redist plists for
+			 * bgp-direct.
+			 * If we later add plist support for
+			 * redistributing other
+			 * protocols, we'll need to loop over protocols
+			 * here
+			 */
+			if (rfg->plist_redist_name
+			    [ZEBRA_ROUTE_BGP_DIRECT][afi]) {
 				vty_out(vty,
-					"  redistribute bgp-direct route-map %s\n",
-					rfg->routemap_redist_name
-						[ZEBRA_ROUTE_BGP_DIRECT]);
+					"  redistribute bgp-direct %s prefix-list %s\n",
+					afistr,
+					rfg->plist_redist_name
+					[ZEBRA_ROUTE_BGP_DIRECT]
+					[afi]);
 			}
-			if (rfg->routemap_redist_name
-				    [ZEBRA_ROUTE_BGP_DIRECT_EXT]) {
+			if (rfg->plist_redist_name
+			    [ZEBRA_ROUTE_BGP_DIRECT_EXT][afi]) {
```

Comment:
I'm not a fan of line breaks before indexes -- better to remove whitespace on the indent than to add white space that has semantic implications.